# Examining Racial Discrimination in the US Job Market

### Background
Racial discrimination continues to be pervasive in cultures throughout the world. Researchers examined the level of racial discrimination in the United States labor market by randomly assigning identical résumés to black-sounding or white-sounding names and observing the impact on requests for interviews from employers.

### Data
In the dataset provided, each row represents a resume. The 'race' column has two values, 'b' and 'w', indicating black-sounding and white-sounding. The column 'call' has two values, 1 and 0, indicating whether the resume received a call from employers or not.

Note that the 'b' and 'w' values in race are assigned randomly to the resumes when presented to the employer.

<div class="span5 alert alert-info">
### Exercises
You will perform a statistical analysis to establish whether race has a significant impact on the rate of callbacks for resumes.

Answer the following questions **in this notebook below and submit to your Github account**. 

   1. What test is appropriate for this problem? Does CLT apply?
   2. What are the null and alternate hypotheses?
   3. Compute margin of error, confidence interval, and p-value. Try using both the bootstrapping and the frequentist statistical approaches.
   4. Write a story describing the statistical significance in the context or the original problem.
   5. Does your analysis mean that race/name is the most important factor in callback success? Why or why not? If not, how would you amend your analysis?

You can include written notes in notebook cells using Markdown: 
   - In the control panel at the top, choose Cell > Cell Type > Markdown
   - Markdown syntax: http://nestacms.com/docs/creating-content/markdown-cheat-sheet


#### Resources
+ Experiment information and data source: http://www.povertyactionlab.org/evaluation/discrimination-job-market-united-states
+ Scipy statistical methods: http://docs.scipy.org/doc/scipy/reference/stats.html 
+ Markdown syntax: http://nestacms.com/docs/creating-content/markdown-cheat-sheet
+ Formulas for the Bernoulli distribution: https://en.wikipedia.org/wiki/Bernoulli_distribution
</div>
****

In [11]:
import pandas as pd
import numpy as np
from scipy import stats

In [2]:
data = pd.io.stata.read_stata('us_job_market_discrimination.dta')

In [26]:
# number of callbacks for black-sounding names
sum(data[data.race=='b'].call)

157.0

In [4]:
data.head()

id ad  education  ofjobs  yearsexp  honors  volunteer  military  empholes  \
0  b  1          4       2         6       0          0         0         1   
1  b  1          3       3         6       0          1         1         0   
2  b  1          4       1         6       0          0         0         0   
3  b  1          3       4         6       0          1         0         1   
4  b  1          3       3        22       0          0         0         0   

   occupspecific    ...      compreq  orgreq  manuf  transcom  bankreal trade  \
0             17    ...          1.0     0.0    1.0       0.0       0.0   0.0   
1            316    ...          1.0     0.0    1.0       0.0       0.0   0.0   
2             19    ...          1.0     0.0    1.0       0.0       0.0   0.0   
3            313    ...          1.0     0.0    1.0       0.0       0.0   0.0   
4            313    ...          1.0     1.0    0.0       0.0       0.0   0.0   

  busservice othservice  missind  ownership  
0        0.0        0.0      0.0             
1        0.0        0.0      0.0             
2        0.0        0.0      0.0             
3        0.0        0.0      0.0             
4        0.0        1.0      0.0  Nonprofit  

[5 rows x 65 columns]

In [5]:
data.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 4870 entries, 0 to 4869
Data columns (total 65 columns):
id                    4870 non-null object
ad                    4870 non-null object
education             4870 non-null int8
ofjobs                4870 non-null int8
yearsexp              4870 non-null int8
honors                4870 non-null int8
volunteer             4870 non-null int8
military              4870 non-null int8
empholes              4870 non-null int8
occupspecific         4870 non-null int16
occupbroad            4870 non-null int8
workinschool          4870 non-null int8
email                 4870 non-null int8
computerskills        4870 non-null int8
specialskills         4870 non-null int8
firstname             4870 non-null object
sex                   4870 non-null object
race                  4870 non-null object
h                     4870 non-null float32
l                     4870 non-null float32
call                  4870 non-null float32
city        

In [6]:
data.describe()

education       ofjobs     yearsexp       honors    volunteer  \
count  4870.000000  4870.000000  4870.000000  4870.000000  4870.000000   
mean      3.618480     3.661396     7.842916     0.052772     0.411499   
std       0.714997     1.219126     5.044612     0.223601     0.492156   
min       0.000000     1.000000     1.000000     0.000000     0.000000   
25%       3.000000     3.000000     5.000000     0.000000     0.000000   
50%       4.000000     4.000000     6.000000     0.000000     0.000000   
75%       4.000000     4.000000     9.000000     0.000000     1.000000   
max       4.000000     7.000000    44.000000     1.000000     1.000000   

          military     empholes  occupspecific   occupbroad  workinschool  \
count  4870.000000  4870.000000    4870.000000  4870.000000   4870.000000   
mean      0.097125     0.448049     215.637782     3.481520      0.559548   
std       0.296159     0.497345     148.127551     2.038036      0.496492   
min       0.000000     0.000000       7.000000     1.000000      0.000000   
25%       0.000000     0.000000      27.000000     1.000000      0.000000   
50%       0.000000     0.000000     267.000000     4.000000      1.000000   
75%       0.000000     1.000000     313.000000     6.000000      1.000000   
max       1.000000     1.000000     903.000000     6.000000      1.000000   

          ...           educreq      compreq       orgreq        manuf  \
count     ...       4870.000000  4870.000000  4870.000000  4870.000000   
mean      ...          0.106776     0.437166     0.072690     0.082957   
std       ...          0.308866     0.496083     0.259649     0.275854   
min       ...          0.000000     0.000000     0.000000     0.000000   
25%       ...          0.000000     0.000000     0.000000     0.000000   
50%       ...          0.000000     0.000000     0.000000     0.000000   
75%       ...          0.000000     1.000000     0.000000     0.000000   
max       ...          1.000000     1.000000     1.000000     1.000000   

          transcom     bankreal        trade   busservice   othservice  \
count  4870.000000  4870.000000  4870.000000  4870.000000  4870.000000   
mean      0.030390     0.085010     0.213963     0.267762     0.154825   
std       0.171677     0.278932     0.410141     0.442847     0.361773   
min       0.000000     0.000000     0.000000     0.000000     0.000000   
25%       0.000000     0.000000     0.000000     0.000000     0.000000   
50%       0.000000     0.000000     0.000000     0.000000     0.000000   
75%       0.000000     0.000000     0.000000     1.000000     0.000000   
max       1.000000     1.000000     1.000000     1.000000     1.000000   

           missind  
count  4870.000000  
mean      0.165092  
std       0.371308  
min       0.000000  
25%       0.000000  
50%       0.000000  
75%       0.000000  
max       1.000000  

[8 rows x 55 columns]

In [9]:
data.shape

(4870, 65)

<div class="span5 alert alert-success">
<p>Your answers to Q1 and Q2 here</p>
</div>

**Answer to Q1:** Chi-squared test and two-sample z tests are appropriate for this. 
Yes, the CLT applies here because we can assume the samples are representative of the population and are independent.

**Answer to Q2:** The null hypothesis is that there is no difference between the black-sounding and white-sounding names in terms of callback.
Therefore the alternative hypothesis is that there is a difference.

**Answers to Q3 follow...**

In [10]:
w = data[data.race=='w']
b = data[data.race=='b']

In [25]:
prop_w = np.sum(w.call) / len(w)
prop_b = np.sum(b.call) / len(b)

prop_diff = prop_w - prop_b
phat = (np.sum(w.call) + np.sum(b.call)) / (len(w) + len(b))

z = prop_diff / np.sqrt(phat * (1 - phat) * ((1 /len(w)) + (1 / len(b))))
pval = stats.norm.cdf(-z) * 2
print("Z score: {}".format(z))
print("P-value: {}".format(pval))

Z score: 4.108412152434346
P-value: 3.983886837585077e-05


In [29]:
#bootstrap approach

perm_reps = np.empty(100000)

white = data[data.race == 'w'].call.values
black = data[data.race == 'b'].call.values
diff_of_means = np.mean(white) - np.mean(black)

for i in range(len(perm_reps)):
    permutation_samples = np.random.permutation(np.concatenate((white, black)))
    
    white_perm = permutation_samples[:len(white)]
    black_perm = permutation_samples[len(white):]
    
    perm_reps[i] = np.abs(np.mean(white_perm) - np.mean(black_perm))

p = np.sum(perm_reps > diff_of_means) / len(perm_reps)
print('p: {}'.format(p))

p: 2e-05


In [32]:
errormargin = 1.96 * np.sqrt(phat * (1 - phat) * ((1 / len(w) + (1 / len(b)))))
confint = prop_diff + np.array([-1, 1]) * errormargin
print("Margin of Error: {}".format(errormargin))
print("Confidence interval: {}".format(confint))

Margin of Error: 0.015281912310894095
Confidence interval: [0.01675094 0.04731477]


In [33]:
# Chi-square approach:
cont_table = pd.crosstab(index=data.call, columns=data.race)
chi2, pval, _, _ = stats.chi2_contingency(cont_table)
print("Chi-squared test statistic: {}".format(chi2))
print("p-value: {}".format(pval))

Chi-squared test statistic: 16.44902858418937
p-value: 4.997578389963255e-05


<div class="span5 alert alert-success">
<p> Your answers to Q4 and Q5 here </p>
</div>

**Answer to Q4:** The p-value is very, very small (2e-05)--far less than 0.05--suggesting that it is statistically significant and that we should reject the null hypothesis that white- and black-sounding names have the same callback rate. Further, since 0 is outside our confidence interval of 0.0168 and 0.0473, we have more reason to justify rejecting the null hypothesis.

**Answer to Q5:** No, we cannot/should not conclude that race (or race-sounding names) is *the most* important factor in determining if someone is likely to get a callback; it is simply *a* statistically significant factor. There could be a number of other factors or compounding factors. These need to be considered and further analyses conducted.